src/Application/Items/Commands/UpgradeUserItemCommand.cs

```diff
@@ -0,0 +1,109 @@
+using AutoMapper;
+using Crpg.Application.Common;
+using Crpg.Application.Common.Interfaces;
+using Crpg.Application.Common.Mediator;
+using Crpg.Application.Common.Results;
+using Crpg.Application.Common.Services;
+using Crpg.Application.Items.Models;
+using Crpg.Domain.Entities.Items;
+using Crpg.Domain.Entities.Users;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using NetTopologySuite.Index.HPRtree;
+using LoggerFactory = Crpg.Logging.LoggerFactory;
+
+namespace Crpg.Application.Items.Commands;
+
+public record UpgradeUserItemCommand : IMediatorRequest<UserItemViewModel>
+{
+    public int UserItemId { get; init; }
+    public int UserId { get; init; }
+
+    internal class Handler : IMediatorRequestHandler<UpgradeUserItemCommand, UserItemViewModel>
+    {
+        private static readonly ILogger Logger = LoggerFactory.CreateLogger<UpgradeUserItemCommand>();
+
+        private readonly ICrpgDbContext _db;
+        private readonly IMapper _mapper;
+        private readonly IActivityLogService _activityLogService;
+
+        public Handler(ICrpgDbContext db, IMapper mapper, IActivityLogService activityLogService)
+        {
+            _db = db;
+            _mapper = mapper;
+            _activityLogService = activityLogService;
+        }
+
+        public async Task<Result<UserItemViewModel>> Handle(UpgradeUserItemCommand req, CancellationToken cancellationToken)
+        {
+            var userItem = await _db.UserItems
+                .Include(ui => ui.User!)
+                .Include(ui => ui.Item!)
+                .FirstOrDefaultAsync(ui => ui.UserId == req.UserId && ui.Id == req.UserItemId, cancellationToken);
+
+            var user = await _db.Users
+                .Include(u => u.Items)
+                .FirstOrDefaultAsync(u => u.Id == req.UserId, cancellationToken);
+            if (user == null)
+            {
+                return new(CommonErrors.UserNotFound(req.UserId));
+            }
+
+            var userUserItems = await _db.UserItems
+                .Where(ui => user.Items.Contains(ui))
+                .Include(ui => ui.Item)
+                .ToListAsync();
```

Comment:
I'm not sure what this code is doing. Is it reloading the user items again?

Reply:
I needed to include the Item , but indeed there was no need to query the database twice.